src/gui/tray/UserModel.cpp

```diff
@@ -104,7 +93,7 @@ void User::slotBuildNotificationDisplay(const ActivityList &list)
             continue;
         }
         const auto message = AccountManager::instance()->accounts().count() == 1 ? "" : activity._accName;
-        showOsNotification(activity._subject, message);
+        showDesktopNotification(activity._subject, message);
```

Comment:
@FlexW To mitigate risks of having a collision, I would also think of adding **_accName** into the hash calculation. For those cases when a user has multiple accounts added.

Reply:
This is a good point, actually it occurred to me that the whole approach based on formerly QSet<int> and now set<uint> is in fact weak behavior wise. It's doomed to generate collisions, if you look at qHash implementation it's a hash function to try to balance the buckets in a hash map, it's not something stronger about "do I already have seen this exact string before". With a QSet you're oblivious to such collisions and so the chances are high that some messages are being eaten up even today.

Now it's probably not the role of that PR to fix this, It's clearly too big already this was supposedly about adding another warning type and it's now going to refactor the whole notification handling in here. Would have benefited from being two PRs (one for said warning, and another to rework the desktop notification caching). There's plenty of time in the 3.3 cycle to do this and backport one or both of them on stable-3.2.

But clearly with the discussions around that topic that we had on this PR, the whole use of QSet/set is something to keep an eye on and *maybe* fix. I mean in a way people hate desktop notifications it's not like they're going to shout if some are not happening... OK... apart the ones from Talk those they should always fire or complaints will happen.